sasmodels/direct_model.py

```diff
@@ -456,43 +453,137 @@ def test_reparameterize():
     except Exception:
         pass
 
+def _direct_calculate(model, data, pars):
+    from .core import load_model_info, build_model
+    model_info = load_model_info(model)
+    kernel = build_model(model_info)
+    calculator = DirectModel(data, kernel)
+    return calculator(**pars)
+
+def Iq(model, q, dq=None, ql=None, qw=None, **pars):
+    """
+    Compute I(q) for *model*. Resolution is *dq* for pinhole or *ql* and *qw*
+    for slit geometry. Use 0 or None for infinite slits.
```

Comment:
When I tested the Iq function with an infinite slit I ran into a couple errors. When qw=None, ql is still ignored and slit resolution is not applied. When using 0 I get an error due to the indexing on line 262-263. I made code suggestions in a few places that I think will fix this but need review. I made the assumption that the dxl and dxw attributes of Data1D should be kept as either None or an array of same length as q.